Automattic/wp-calypso
# Checkout: Refactor the 'compareTo' price calculation for product variations.

This PR is an improvement on https://github.com/Automattic/wp-calypso/pull/84429



## Proposed Changes

* Re-introduced the code block specifically for first-year introductory discounts. (For all product types).
* Removed the code block specifying a compareTo price specifically for Jetpack & Akismet products. Instead I improved the calculation so that it works for All for Biennial, Triennial, Quadrennial, etc. products.


## Testing Instructions

- Checkout this PR branch and yarn start OR click the Calypso Live (direct link) below.
- Open checkout with either a Jetpack, or Akismet, or WPCOM product. (a monthly billing-term product)
- Compare with production, each product you test.
- Once checkout is open with a monthly product in the cart, open the billing-term variation dropdown and verify the "Save x%" on all variations (monthly, yearly, bi-yearly, etc) looks correct, and they are unchanged from production.




## Pre-merge Checklist
